sdk/azcore/policy_retry.go

```diff
@@ -160,9 +161,12 @@ func (p *retryPolicy) Do(ctx context.Context, req *Request) (resp *Response, err
 		}
 		defer rwbody.realClose()
 	}
-	for try := int32(1); try <= p.options.MaxTries; try++ {
+	try := int32(1)
+	for {
 		resp = nil // reset
-		logf("\n=====> Try=%d\n", try)
+		if Log().Should(LogRetry) {
```

Comment:
You could capture the result of this once at the top of the method and then not call Should() multiple times in this method.

Reply:
Will refactor however I'd be surprised if the compiler didn't hoist this call out of the for body (loop invariant code motion).